src/nano_contracts/builder.ts

```diff
@@ -439,17 +544,35 @@ class NanoContractTransactionBuilder {
       tokens = Array.from(tokenSet);
       for (const action of this.actions) {
         // Call action
-        if (action.type === NanoContractActionType.DEPOSIT) {
-          const ret = await this.executeDeposit(action, tokens);
-          inputs = concat(inputs, ret[0]);
-          outputs = concat(outputs, ret[1]);
-        } else if (action.type === NanoContractActionType.WITHDRAWAL) {
-          const output = this.executeWithdrawal(action, tokens);
-          if (output) {
-            outputs = concat(outputs, output);
+        switch (action.type) {
+          case NanoContractActionType.DEPOSIT: {
+            const retDeposit = await this.executeDeposit(action);
+            inputs = concat(inputs, retDeposit[0]);
+            outputs = concat(outputs, retDeposit[1]);
```

Comment:
Instead of `[inputs, outputs]` and accessing inputs/outputs by index why not return `{ inputs, outputs }` so we can use normal destructuring?